core/domain/classifier_services.py

```diff
@@ -390,45 +394,160 @@ def get_classifier_training_job_by_id(job_id):
+        state_name: str. The name of the state to which the classifier
+            belongs.
+        training_data: dict. The data used in training phase.
+        status: str. The status of the training job (NEW by default).
+
+    Returns:
+        job_id: str. ID of the classifier training job.
+    """
+    next_scheduled_check_time = datetime.datetime.utcnow()
+    dummy_classifier_training_job = classifier_domain.ClassifierTrainingJob(
+        'job_id_dummy', algorithm_id, interaction_id, exp_id, exp_version,
+        next_scheduled_check_time, state_name, status, training_data)
+    dummy_classifier_training_job.validate()
+    job_id = classifier_models.ClassifierTrainingJobModel.create(
+        algorithm_id, interaction_id, exp_id, exp_version,
+        next_scheduled_check_time, training_data, state_name, status)
+    return job_id
+
+
+def _update_classifier_training_jobs_status(job_ids, status):
     """Checks for the existence of the model and then updates it.
 
     Args:
-        job_id: str. ID of the ClassifierTrainingJob domain object.
+        job_id: list(str). list of ID of the ClassifierTrainingJob domain
+            objects.
         status: str. The status to which the job needs to be updated.
 
     Raises:
         Exception. The ClassifierTrainingJobModel corresponding to the job_id
             of the ClassifierTrainingJob does not exist.
     """
-    classifier_training_job_model = (
-        classifier_models.ClassifierTrainingJobModel.get(job_id, strict=False))
-    if not classifier_training_job_model:
-        raise Exception(
-            'The ClassifierTrainingJobModel corresponding to the job_id of the'
-            'ClassifierTrainingJob does not exist.')
+    classifier_training_job_models = (
+        classifier_models.ClassifierTrainingJobModel.get_multi(job_ids))
 
-    initial_status = classifier_training_job_model.status
-    if status not in feconf.ALLOWED_TRAINING_JOB_STATUS_CHANGES[initial_status]:
-        raise Exception(
-            'The status change %s to %s is not valid.' % (initial_status,
-                                                          status))
+    for index, job_id in enumerate(job_ids):
+        if classifier_training_job_models[index] is None:
+            raise Exception(
+                'The ClassifierTrainingJobModel corresponding to the job_id '
+                'of the ClassifierTrainingJob does not exist.')
 
-    classifier_training_job = get_classifier_training_job_by_id(job_id)
-    classifier_training_job.update_status(status)
-    classifier_training_job.validate()
+        initial_status = classifier_training_job_models[index].status
+        if status not in (
+                feconf.ALLOWED_TRAINING_JOB_STATUS_CHANGES[initial_status]):
+            raise Exception(
+                'The status change %s to %s is not valid.' % (
+                    initial_status, status))
+
+        classifier_training_job = get_classifier_training_job_by_id(
```

Comment:
Hang on a sec. Why are we retrieving the model again? (That's what get_..._by_id() does.) We already have the models. Can't we just convert those to domain objects directly?